src/Resources/views/CRUD/base_array_macro.html.twig

```diff
@@ -8,17 +8,42 @@ For the full copyright and license information, please view the LICENSE
 file that was distributed with this source code.
 
 #}
-{% macro render_array(value, inline) %}
-    {% from _self import render_array %}
-    {% for key, val in value %}
-        {% if val is iterable %}
-            [{{ key }} => {{ render_array(val, inline) }}]
-        {% else %}
-            [{{ key }} => {{ val }}]
-        {% endif %}
+{% macro render_array(value, inline, values_only, translate, translation_domain, translation_prefix, translate_key) %}
+    {% spaceless %}
+        {% if not inline %}<ul>{% endif %}
+        {% from _self import render_array %}
+        {% for key, val in value %}
+            {% if inline == true %}
+                [
+            {% else %}
+                {% if val is not iterable or not values_only %}
+                    <li>
+                {% endif %}
+            {% endif %}
 
-        {% if not loop.last and not inline %}
-            <br>
-        {% endif %}
-    {% endfor %}
+            {% if not values_only %}
+                {{ key }} =>
+            {% endif %}
+
+            {% if val is iterable %}
+                {{ render_array(val, inline, values_only, translate, translation_domain, translation_prefix, translate_key) }}
+            {% else %}
+                {% if translate == true %}
+                    {% set translate_label = translation_prefix ~ (translate_key == 'key' ? key : val) %}
```

Comment:
I feel like there is too many `translate`-specific options. 
And I kinda disagree with the option `translation_prefix` since it doesn't even exist in the classic `show_trans`.
![image](https://user-images.githubusercontent.com/9052536/78583335-00318080-7837-11ea-8bed-c1d90901aee9.png)

I would recommend to use a custom getter with an array_map instead if a prefix or a modification of the key is needed.

Plus, when I saw `translate_key`, I thought it gonna translate the keys.

Nice idea for the translation_domain.
But WDYT about just: 
```
$listMapper
            ->add('options', 'array', [
                'inline' => true,
                'key_translation_domain' => true/false/null/string,
                'values_translation_domain' => true/false/null/string,
            ])
```
It would be similar to https://symfony.com/doc/current/reference/forms/types/choice.html#choice-translation-domain

It would be `false` by default, and if you provide something, it would be translated. So we don't need both `translate` and `translation_domain` options.

Reply:
It looks pretty good, thank you! As I understand we should use `key_translation_domain` or `values_translation_domain` and don't use both at the same time, but what if for some reason will be defined both?

Reply:
You can use both.
`key_translation_domain` is for the key.
`value_translation_domain` is for the value.


Since we displayed before
```
key => value
```
it would be now
```
key|trans({}, key_translation_domain) => value|trans({}, value_translation_domain)
```

Reply:
Got it, thanks

Reply:
My only concern is that I don't know how it will be user-friendly that
`value_translation_domain => true` will use the admin.translation_domain
and
`value_translation_domain => null` will use the `'messages'` default translation_domain.

Reply:
Improved